common_package_py/commands.py

```diff
@@ -0,0 +1,344 @@
+        any `number*` data type will always return true.
+
+        Note:
+            Also returns true if the json_iter has a type different from `number*`
+
+        Args:
+            json_iter (any): The value of the JSON key.
+
+        Returns:
+            bool: True if the value is within the bounds, False otherwise.
+        """
+
+        jsk_type_check = JsonKeyDefinition(False, {data_type.NUMBER})
+
+        if data_type.NUMBER in self.data_types or \
+                data_type.NUMBER_FLOAT in self.data_types or \
+            data_type.NUMBER_INTEGER in self.data_types or \
+                data_type.NUMBER_UNSIGNED in self.data_types:
+
+            if self.min_val is not None and jsk_type_check.type_check(json_iter) and json_iter < self.min_val:
+                return False
+            if self.max_val is not None and jsk_type_check.type_check(json_iter) and json_iter > self.max_val:
+                return False
+        return True
+
+    def type_check(self, json_iter: any) -> bool:
+        """
+        Checks if the value of the JSON key matches any of the allowed data types.
+
+        Args:
+            json_iter (any): The value of the JSON key.
+
+        Returns:
+            bool: True if the value matches any of the allowed data types, False otherwise.
+        """
+
+        for data_type in self.data_types:
+            match data_type:
+                case data_type.NULL:
+                    if json_iter is None:
+                        return True
+                case data_type.BOOLEAN:
+                    if isinstance(json_iter, bool):
+                        return True
+                case data_type.NUMBER:
+                    if isinstance(json_iter, (int, float)) and not isinstance(json_iter, (bool)):
+                        return True
+                case data_type.NUMBER_FLOAT:
+                    if isinstance(json_iter, float) and not isinstance(json_iter, (bool)):
+                        return True
+                case data_type.NUMBER_INTEGER:
+                    if isinstance(json_iter, int) and not isinstance(json_iter, (bool)):
+                        return True
+                case data_type.NUMBER_UNSIGNED:
+                    if isinstance(json_iter, int) and json_iter >= 0 and not isinstance(json_iter, (bool)):
+                        return True
+                case data_type.STRING:
+                    if isinstance(json_iter, str):
+                        return True
+                case data_type.ARRAY:
+                    if isinstance(json_iter, list):
+                        return True
+                case data_type.OBJECT:
+                    if isinstance(json_iter, dict):
+                        return True
+                case _:
+                    raise RuntimeError(
+                        f"JsonKeyDefinition::type_check: Unknown data_type provided: {data_type}")
+
+        return False
+
+    def __eq__(self, other: 'JsonKeyDefinition') -> bool:
```

Comment:
Self nutzen
Je nach Python version die ihr nutzt ist das in anderen Paketen.
```suggestion
    def __eq__(self, other: Self) -> bool:
```

Reply:
Der findet den Import überhaupt nicht und auch sonst ist das ab Python3.11 schon wieder wo anders.
Das ist eine absolute Katastrophe mit Self und ich lasse das jetzt einfach so, wie es ist.

Quelle:
https://realpython.com/python-type-self/#how-to-annotate-a-method-with-the-self-type-in-python:~:text=from%20typing_extensions%20import%20Self (funzt nur nicht trotz Python 3.10)